covidwatchorg/portal
# non-bootstrap mobile refactor for Login / NavBar / Footer

### include-media
A potential direction for the mobile refactor that makes use of css media queries, rather than retrofitting the codebase with Bootstrap. 

I found a nice css library called [include-media](https://eduardoboucas.github.io/include-media) that makes the breakpoints easier to read and keep consistent, i.e. we can set the following in `_include-media.scss`
```scss
// _include-media.scss

$breakpoints: (
  'phone': 420px,
  'tablet': 768px,
  'desktop': 1024px,
) !default;
```
so that if we write
```scss
@include media('<=phone') {
    margin: 0 5px;
}
```
it compiles down to
```css
@media (max-width: 420px) {
    margin: 0 5px;
}
```

### Login
The login screen on mobile should look something like the following
<img width="357" alt="Screen Shot 2020-08-05 at 2 34 24 PM" src="https://user-images.githubusercontent.com/13578537/89461288-f35f6380-d728-11ea-8f1d-861c4a6d1e8c.png">

Rather than deal with trying to fit the big welcome blurb and picture of students hanging out with each other into this tiny interface, I elected to just set `display: none;` and eliminate them.

### Footer
The footer scss code is a bit convoluted, but it appears to get the job done. The key bit of trickery is
```scss
@include media('>phone') {
    flex-direction: row;
    ...
}
  @include media('<=phone') {
    flex-direction: column;
}
```
which ensures the footer stack horizontally on desktop, and vertically on mobile.

### NavBar
For the NavBar I made the call that it would be simplest to just remove the avatar/name/role section of the display (when logged in). So for desktop you should have something like this:
<img width="1257" alt="Screen Shot 2020-08-05 at 2 41 16 PM" src="https://user-images.githubusercontent.com/13578537/89461904-dd05d780-d729-11ea-8b31-96b6b0b65860.png">
which should become this on mobile:
<img width="355" alt="Screen Shot 2020-08-05 at 2 42 54 PM" src="https://user-images.githubusercontent.com/13578537/89461974-f7d84c00-d729-11ea-9e02-0fcfda190bc0.png">

### To Test
The dev tools are an obvious place to start. I also found it useful to expose the app to the internet using `ngrok`
```
ngrok http 3000 -host-header="localhost:3000"
```
and check it out on my phone, since I've found that sometimes (often) there are discrepancies between the display in dev tools and things look on an actual mobile device.
